bin/site-inspector

```diff
@@ -1,15 +1,29 @@
 #!/usr/bin/env ruby
 
 require_relative "../lib/site-inspector"
+
 require "oj"
 
 domain = ARGV[0]
+http_mode = (ARGV[1] == "--http")
 
 if domain.to_s.empty?
-  puts "Usage: site-inspector [DOMAIN]"
+  puts "Usage: site-inspector [DOMAIN] [--http]"
   exit 1
 end
 
-details = SiteInspector.new(domain).to_hash
+# HTTP mode:
+#   * all details for possible endpoints
+#   * don't follow redirects
+#   * shorter timeout
+if http_mode
+  site = SiteInspector.new(domain, {:timeout => 5})
+  details = site.http
+
```

Comment:
HTTP mode uses a timeout of 5 seconds, since it makes so many more HTTP requests (though fewer DNS requests) than normal mode.